ejabberd_bridge/management/commands/ejabberd_auth.py

```diff
@@ -100,18 +105,20 @@ def handle(self, *args, **options):
             while True:
                 data = self.from_ejabberd()
                 self.logger.debug("Command is %s" % data[0])
-                if not isinstance(data[1], (int, long)):
-                    success = False
 
-                if data[0] == "auth":
-                    success = self.auth(data[1], data[2], data[3])
-                elif data[0] == "isuser":
-                    success = self.isuser(data[1], data[2])
-                elif data[0] == "setpass":
-                    success = False
+                if self.validate_data_from_ejabberd(data):
+                    if data[0] == "auth":
+                        success = self.auth(data[1], data[2], data[3])
+                    elif data[0] == "isuser":
+                        success = self.isuser(data[1], data[2])
+                    elif data[0] == "setpass":
+                        success = False
                 self.to_ejabberd(success)
                 if not options.get("run_forever", True):
                     break
+        except AuthenticationFailed:
+            pass
         except Exception as e:
             self.logger.error("An error has occurred during eJabberd external authentication: %s" % e)
+        finally:
```

Comment:
@fichie23 why use finally here? in case there's no exception, it will also be executed, which mean the `self.to_ejabberd` to be called twice

Reply:
ah sure sorry for missing that